ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -283,61 +303,36 @@ def take_step(
         # If any lessons were incremented or the environment is
         # ready to be reset
         if self.meta_curriculum and any(lessons_incremented.values()):
-            curr_info = self._reset_env(env)
+            self._reset_env(env)
             for brain_name, trainer in self.trainers.items():
                 trainer.end_episode()
             for brain_name, changed in lessons_incremented.items():
                 if changed:
                     self.trainers[brain_name].reward_buffer.clear()
 
-        # Decide and take an action
-        take_action_vector = {}
-        take_action_memories = {}
-        take_action_text = {}
-        take_action_value = {}
-        take_action_outputs = {}
-        for brain_name, trainer in self.trainers.items():
-            action_info = trainer.get_action(curr_info[brain_name])
-            take_action_vector[brain_name] = action_info.action
-            take_action_memories[brain_name] = action_info.memory
-            take_action_text[brain_name] = action_info.text
-            take_action_value[brain_name] = action_info.value
-            take_action_outputs[brain_name] = action_info.outputs
         time_start_step = time()
-        new_info = env.step(
-            vector_action=take_action_vector,
-            memory=take_action_memories,
-            text_action=take_action_text,
-            value=take_action_value,
-        )
+        new_step_infos = env.step()
         delta_time_step = time() - time_start_step
+
+        for step_info in new_step_infos:
+            for brain_name, trainer in self.trainers.items():
+                if brain_name in self.trainer_metrics:
+                    self.trainer_metrics[brain_name].add_delta_step(delta_time_step)
```

Comment:
Note - I think you're double counting the delta_time_step. But I also think the TrainerMetrics code is broken because add_delta_step() increments the experience collection counter 🤦‍♂  https://github.com/Unity-Technologies/ml-agents/blob/ea60bc9089daa6064763463931ef6b4843d2ac4c/ml-agents/mlagents/trainers/trainer_metrics.py#L59-L66
Should be able to clean this up with the timer code soon.